controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go

```diff
@@ -53,5 +55,48 @@ func init() {
 
 // KubeadmControlPlaneTemplateResource describes the data needed to create a KubeadmControlPlane from a template.
 type KubeadmControlPlaneTemplateResource struct {
-	Spec KubeadmControlPlaneSpec `json:"spec"`
+	Spec KubeadmControlPlaneTemplateResourceSpec `json:"spec"`
+}
+
+// KubeadmControlPlaneTemplateResourceSpec defines the desired state of KubeadmControlPlane.
+// NOTE: KubeadmControlPlaneTemplateResourceSpec is similar to KubeadmControlPlaneSpec but
+// omits Replicas and Version fields. These fields do not make sense on the KubeadmControlPlaneTemplate,
+// because they are calculated by the Cluster topology reconciler during reconciliation and thus cannot
+// be configured on the KubeadmControlPlaneTemplate.
+type KubeadmControlPlaneTemplateResourceSpec struct {
+	// MachineTemplate contains information about how machines
+	// should be shaped when creating or updating a control plane.
+	// +optional
+	MachineTemplate *KubeadmControlPlaneTemplateMachineTemplate `json:"machineTemplate,omitempty"`
```

Comment:
@vincepri @CecileRobertMichon 

I did another manual test. This time I tried not setting machineTemplate at all, so I noticed that machineTemplate was still mandatory.

I think we shouldn't force users to specify an empty machineTemplate object, so I made it optional

Reply:
/lgtm